helper/locksutil/locks.go

```diff
@@ -0,0 +1,12 @@
+package locksutil
+
+import (
+	"fmt"
+	"sync"
+)
+
+func CreateLocks(p map[string]*sync.RWMutex, count int64) {
+	for i := int64(0); i < count; i++ {
```

Comment:
Given that you can only have a max of 256 when using `%02` I'd suggest adding a guard here on the size of `count`, and returning an error if too big. We'd catch that in dev, but better to do so than not.